### PR TITLE
Added config option SetPrivateReposAsDefault

### PR DESCRIPTION
**Pull request by manojmalik20**

### CODE REVIEW

File: server/plugin/command.go [hunk 1]
```diff
@@ -470,7 +470,14 @@ func (p *Plugin) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (*mo
 		}
 
 		privateAllowed := false
-		if len(parameters) > 0 {
+		if p.getConfiguration().SetPrivateAsDefault && len(parameters) > 0 {
+			p.postCommandResponse(args, fmt.Sprintf("Unknown command `%v`. Do you meant `/github connect`?", args.Command))
+			return &model.CommandResponse{}, nil
```
Comment (chetanyakan) [inline server/plugin/command.go]: Do we want to allow connecting with only public repos once private repos are enabled by default?

**Review by chetanyakan — CHANGES_REQUESTED** [review]

File: server/plugin/command.go (hunk 1 above)
Comment (manojmalik20) [reply server/plugin/command.go]: No, it should connect with both public and private repos without writing private in slash command for private repos.

File: server/plugin/manifest.go [hunk 2]
```diff
@@ -108,6 +108,14 @@ const manifestStr = `
         "placeholder": "",
         "default": null
       },
+      {
+        "key": "ConnectToPrivateByDefault",
+        "display_name": "Connect to private Repositories by default:",
+        "type": "bool",
+        "help_text": "(Optional) When enabled, /github connect command will let users connect to their github account and gain access to private repositories without explicitly mentioning private.",
+        "placeholder": "",
```
Comment (hrishavkmr) [inline server/plugin/manifest.go:116]: placeholder field can be removed if it is not required.

File: server/plugin/api.go [hunk 3]
```diff
@@ -481,8 +481,11 @@ func (p *Plugin) getConnected(w http.ResponseWriter, r *http.Request) {
 
 		// Inform the user once that private repositories enabled
 		if val == nil {
-			p.CreateBotDMPost(info.UserID, "Private repositories have been enabled for this plugin. To be able to use them you must disconnect and reconnect your GitHub account. To reconnect your account, use the following slash commands: `/github disconnect` followed by `/github connect private`.", "")
-
+			if config.ConnectToPrivateByDefault {
+				p.CreateBotDMPost(info.UserID, "Private repositories have been enabled for this plugin. To be able to use them you must disconnect and reconnect your GitHub account. To reconnect your account, use the following slash commands: `/github disconnect` followed by `/github connect`.", "")
+			} else {
+				p.CreateBotDMPost(info.UserID, "Private repositories have been enabled for this plugin. To be able to use them you must disconnect and reconnect your GitHub account. To reconnect your account, use the following slash commands: `/github disconnect` followed by `/github connect private`.", "")
+			}
```
Comment (hrishavkmr) [inline server/plugin/api.go]: Both the messages seem similar. can we instead use a single message and then use
```
message := 'Private repo .... by %s';
fmt.Sprintf(message, '`/github connect private`')
fmt.Sprintf(message, '`/github connect`')
```

**Comment by manojmalik20** [comment]
Corresponding PR to mattermost has been made.
https://github.com/mattermost/mattermost-plugin-github/pull/448

**Pull request closed without merging**